apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -701,12 +701,40 @@ static CodeableConcept createCodeableConcept(
    *     Reference#getIdentifier()}
    * @param identifierValue the {@link Identifier#getValue()} to use in {@link
    *     Reference#getIdentifier()}
+   * @param providerIdentifiers the {@link Identifier#getServiceProviderIdentifiers()} to use in
+   *     {@link Reference#getIdentifier()}
    * @return a {@link Reference} with the specified {@link Identifier}
    */
-  static Reference createIdentifierReference(String identifierSystem, String identifierValue) {
-    return new Reference()
-        .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue))
-        .setDisplay(retrieveNpiCodeDisplay(identifierValue));
+  static Reference createIdentifierReference(
+      String identifierSystem,
+      String identifierValue,
+      ServiceProviderIdentifiers... providerIdentifiers) {
```

Comment:
Why is `providerIdentifiers` a vararg (why does it allow for zero to many entries)?

If you're just trying to avoid the need to refactor other uses of it, please don't -- this just adds to tech debt, overall, as your code below only supports zero-or-one `providerIdentifiers`, which puts the lie to the method signature. Instead, bite the bullet and make it an `Optional<...>`. If you really want to avoid the extra refactoring, you can always add an overloaded version of this method without the new parameter.